src/wp-includes/media.php

```diff
@@ -1911,6 +1918,37 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds `decoding=async` attribute to an `img` HTML tag.
+ *
+ * @since 6.0.0
+ *
+ * @param string $image         The HTML `img` tag where the attribute should be added.
+ * @param string $context       Additional context to pass to the filters.
+ * @return string Converted `img` tag with `decoding` attribute added.
+ */
+function wp_img_tag_add_decoding_async_attr( $image, $context ) {
+	/**
```

Comment:
I'd suggest adding a guard clause to check if the `decoding` attribute already exists and, if yes, bailing out. Why? To encapsulate the check here where it's most needed to avoid duplication.

```suggestion
	if ( str_contains( $image, ' decoding=' ) ) {
		return $image;
	}
	/**
```

Reply:
We already check that the tag doesn't contain the decoding string above (https://github.com/WordPress/wordpress-develop/pull/2409/files#diff-4d28b6652f0dab208c19aa45e1cf09f3c967060d8cc77502bfcd473a3216b612R1849) - do you still think we need it here @hellofromtonya?